From 8b52a1fbde2d9ef0b2d96017aea952dfae60b2a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Tue, 13 Apr 2021 18:45:11 +0200
Subject: [PATCH] Add launcher for gnome-screensaver

Since GNOME 40 provides a DBus service activation file, usually the DBus path is allocated by gnome-shell before gnome-screensaver runs. This change kills gnome-shell before starting gnome-screensaver, so it can run without problems.
---
 src/session/budgie-desktop-screensaver            | 3 +++
 src/session/budgie-desktop-screensaver.desktop.in | 2 +-
 src/session/meson.build                           | 6 ++++++
 3 files changed, 10 insertions(+), 1 deletion(-)
 create mode 100755 src/session/budgie-desktop-screensaver

diff --git a/src/session/budgie-desktop-screensaver b/src/session/budgie-desktop-screensaver
new file mode 100755
index 00000000..4dc801a7
--- /dev/null
+++ b/src/session/budgie-desktop-screensaver
@@ -0,0 +1,3 @@
+#!/bin/sh
+pkill -f '/usr/bin/gjs /usr/share/gnome-shell/org.gnome.ScreenSaver'
+gnome-screensaver &
diff --git a/src/session/budgie-desktop-screensaver.desktop.in b/src/session/budgie-desktop-screensaver.desktop.in
index 490f97d8..6c8d866c 100644
--- a/src/session/budgie-desktop-screensaver.desktop.in
+++ b/src/session/budgie-desktop-screensaver.desktop.in
@@ -3,7 +3,7 @@ Type=Application
 Name=Screensaver (Budgie Desktop)
 Comment=Launch screensaver and locker program
 Icon=preferences-desktop-screensaver
-Exec=gnome-screensaver
+Exec=budgie-desktop-screensaver
 TryExec=gnome-screensaver
 OnlyShowIn=Budgie;
 NoDisplay=true
diff --git a/src/session/meson.build b/src/session/meson.build
index e2330c77..b6e26859 100644
--- a/src/session/meson.build
+++ b/src/session/meson.build
@@ -129,3 +129,9 @@ custom_target('desktop-file-screensaver',
     install : true,
     install_dir : xdg_appdir,
 )
+
+# Install screensaver launcher
+install_data(
+    'budgie-desktop-screensaver',
+    install_dir: join_paths(get_option('prefix'), get_option('bindir')),
+)
-- 
2.31.1

